hw4/04-templates.js

```diff
@@ -3,62 +3,118 @@ const axios = require('axios');
 
 const app = express();
 const port = process.env.PORT || 5001;
+const url = 'https://restcountries.com/v3.1/all';
+
 
-// Use Pug as the templating engine
 app.set('views', __dirname + '/views');
 app.set('view engine', 'pug');
 
-// REST Countries URL
-const url = 'https://restcountries.com/v3.1/all';
 
-// Add your code here
+let groupBy = function(data, key) {
+  return data.reduce(function(storage, item) {
+    let group = item[key];
+
+    storage[group] = storage[group] || [];
+    storage[group] = Number(storage[group]) + item.population;
+
+    return storage; 
+  }, {});
+};
+
+
+const instructions = [
+  { 'title'     : '/capitals',
+    'params'    : 'capital',
+    'command'   : '`${country.name.official} - ${country.capital}`',
+  },
+  {
+    'title'     : '/populous',
+    'params'    : 'population',
+    'command'   : 'if (country.population >= 50000000) `${country.name.official} - ${country.population}`',
```

Comment:
country.name.common should be used instead of country.name.official

Reply:
Population needs to be formatted using commas like 4,649,660. The API returns us just 4649660 and we need to do additional work to format it.

Reply:
@smah29 Thanks!

@juhiaugustine I missed that in the instructions. Thanks!